website/src/models/challenge.js

```diff
@@ -213,4 +215,37 @@ schema.methods.removeTask = async function challengeRemoveTask (task) {
   }, {multi: true}).exec();
 };
 
+// Unlink challenges tasks (and the challenge itself) from user
+schema.methods.unlinkTasks = async function challengeUnlinkTasks (user, keep) {
+  let challengeId = this._id;
+  let findQuery = {
+    userId: user._id,
+    'challenge.id': challengeId,
+  };
+
+  removeFromArray(user.challenges, challengeId);
+
+  if (keep === 'keep-all') {
+    await Tasks.Task.update(findQuery, {
+      $set: {challenge: {}}, // TODO what about updatedAt?
+    }, {multi: true}).exec();
+
+    await user.save();
+  } else { // keep = 'remove-all'
+    let tasks = await Tasks.Task.find(findQuery).select('_id type completed').exec();
+    let taskPromises = tasks.map(task => {
+      // Remove task from user.tasksOrder and delete them
+      if (task.type !== 'todo' || !task.completed) {
+        removeFromArray(user.tasksOrder[`${task.type}s`], task._id);
+      }
+
+      return task.remove();
+    });
+    user.markModified('tasksOrder');
```

Comment:
@TheHollidayInn last thing before I can merge (btw, very clean PR!): it the markModified necessary here? removeFroArray uses `array.splice` which should be proxied by mongoose